src/web/app/pages-instructor/instructor-session-edit-page/copy-questions-from-other-sessions-modal/copy-questions-from-other-sessions-modal.component.spec.ts

```diff
@@ -17,21 +140,102 @@ describe('CopyQuestionsFromOtherSessionsModalComponent', () => {
+    testFeedbackSessionTabModel1.isTabExpanded = true;
+    testFeedbackSessionTabModel1.hasQuestionsLoaded = true;
+    testFeedbackSessionTabModel1.questionsTableRowModels = [testQuestionToCopyCandidate1, testQuestionToCopyCandidate2];
+    component.feedbackSessionTabModels = [testFeedbackSessionTabModel1];
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback questions failed to load', () => {
+    testFeedbackSessionTabModel1.isTabExpanded = true;
+    testFeedbackSessionTabModel1.hasLoadingFailed = true;
+    component.feedbackSessionTabModels = [testFeedbackSessionTabModel1];
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should load feedback questions', () => {
+    const feedbackQuestions: FeedbackQuestions = {
+      questions: [testFeedbackQuestion1, testFeedbackQuestion2],
+    };
+    spyOn(feedbackQuestionsService, 'getFeedbackQuestions').and.returnValue(of(feedbackQuestions));
+    component.feedbackSessionTabModels = [testFeedbackSessionTabModel1];
+
+    component.loadQuestions(testFeedbackSessionTabModel1);
+
+    expect(component.feedbackSessionTabModels[0].hasQuestionsLoaded).toBeTruthy();
+    expect(component.feedbackSessionTabModels[0].hasLoadingFailed).toBeFalsy();
+    expect(component.feedbackSessionTabModels[0].questionsTableRowModels.length).toBe(2);
+    expect(component.feedbackSessionTabModels[0].questionsTableRowModels[0].question.feedbackQuestionId)
+      .toBe(testFeedbackQuestion1.feedbackQuestionId);
+    expect(component.feedbackSessionTabModels[0].questionsTableRowModels[1].question.feedbackQuestionId)
+      .toBe(testFeedbackQuestion2.feedbackQuestionId);
+  });
+
+  it('should get selected questions', () => {
+    const feedbackQuestions: FeedbackQuestions = {
+      questions: [testFeedbackQuestion1, testFeedbackQuestion2],
+    };
+    spyOn(feedbackQuestionsService, 'getFeedbackQuestions').and.returnValue(of(feedbackQuestions));
+    testQuestionToCopyCandidate1.isSelected = true;
+    testQuestionToCopyCandidate2.isSelected = true;
+    testQuestionToCopyCandidate3.isSelected = true;
+    testFeedbackSessionTabModel1.questionsTableRowModels = [testQuestionToCopyCandidate1, testQuestionToCopyCandidate2];
+    testFeedbackSessionTabModel2.questionsTableRowModels = [testQuestionToCopyCandidate3];
+    component.feedbackSessionTabModels = [testFeedbackSessionTabModel1, testFeedbackSessionTabModel2];
+    fixture.detectChanges();
+
+    const questions: FeedbackQuestion[] = component.getSelectedQuestions();
+
+    expect(component.hasAnyQuestionsToCopySelected).toBeTruthy();
+    expect(questions.length).toBe(3);
+    expect(questions[0].feedbackQuestionId).toBe(testFeedbackQuestion1.feedbackQuestionId);
+    expect(questions[1].feedbackQuestionId).toBe(testFeedbackQuestion2.feedbackQuestionId);
+    expect(questions[2].feedbackQuestionId).toBe(testFeedbackQuestion3.feedbackQuestionId);
+  });
```

Comment:
I think we are missing the case where questions are copied

Reply:
Thanks for pointing that out! I've modified the component tests